modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/components/page-accessibility-tool/components/useFilteredViolations.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {createContext, useContext, useReducer} from 'react';
+
+export const TYPES = {
+	CATEGORY_ADD: 'CATEGORY_ADD',
+	CATEGORY_REMOVE: 'CATEGORY_REMOVE',
+	IMPACT_ADD: 'IMPACT_ADD',
+	IMPACT_REMOVE: 'IMPACT_REMOVE',
+};
+
+const INITIAL_STATE = {
+	selectedCategories: [],
+	selectedImpact: [],
+};
+
+function violationsReducer(state = INITIAL_STATE, action) {
+	let nextState = state;
+
+	switch (action.type) {
+		case TYPES.CATEGORY_ADD:
+			nextState = {
+				...state,
+				selectedCategories: [
+					...state.selectedCategories,
+					action.payload,
+				],
+			};
+			break;
```

Comment:
We must avoid changing the state, we must consider building a new variable and returning shortly thereafter.

```suggestion
	switch (action.type) {
		case TYPES.CATEGORY_ADD:
			return {
				...state,
				selectedCategories: [
					...state.selectedCategories,
					action.payload,
				],
			};
```

Reply:
Good, is there any doc about it?

Reply:
We do not have documentation on this, but ideally reducer are pure functions, if you modify a reference it can suffer some unwanted side effects, the reducer is always called twice by React. See just the example in the React documentation https://reactjs.org/docs/hooks-reference.html#usereducer

Reply:
Done mi patrón